src/app/qgsmaptoolmovefeature.cpp

```diff
@@ -277,16 +279,21 @@ void QgsMapToolMoveFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
         {
           emit messageEmitted( errorMsg, Qgis::MessageLevel::Critical );
           deleteRubberband();
+          vlayer->deleteFeatures( request.filterFids() );
+          vlayer->destroyEditCommand();
           mSnapIndicator->setMatch( QgsPointLocator::Match() );
         }
+        else
+        {
+          vlayer->endEditCommand();
+        }
         if ( !childrenInfoMsg.isEmpty() )
         {
           emit messageEmitted( childrenInfoMsg, Qgis::MessageLevel::Info );
         }
         break;
     }
 
-    vlayer->endEditCommand();
```

Comment:
Good point. If I recall correctly I initially did it this way because without `return;`ing when the copyMoveFeatures fails you get a mismatch in the undo stack and I didn't want to skip emitting the childrenInfoMsg. But I guess it doesn't really make sense to show the message anyway because the copied features are deleted if the action fails.

I changed it back to how it used to be.